sections/Philosophy.tex

```diff
@@ -0,0 +1,60 @@
+%GAMBLE is used to refer to our languange
+\section{Design philosophy}
+For big computations on a computer one must acknowledge the computing power required for such computations.
+The main focus for GAMBLE is to use the computational powers of the GPU to handle such computations without it being inconvenient for the programmer.
+From \myref{sec:state_of_the_art} we know that several libraries and languages allows the programmer to explicitly designate workloads to the GPU however this often requires explicit memory handling as well.
+GAMBLE is meant to be used for computations which has a need for more computational power; due to this the language should not be too unfamiliar.
+Keeping GAMBLE familiar to other languages reduces the time required to familiarize one with GAMBLE.
+This is done by using a C-like syntax, while stripping GAMBLE of things deemed not necessary for computational needs, as will be documented in the rest of this chapter.
+As the GPU is the resource being used to achieve more computing power, the data computed must also be applicable to the niche of the GPU, i.e. the data must be parallelisable.
+This basic need will influence how data is represented, and also puts focus onto matrices and vector calculations, which often can be paralellised.
```

Comment:
Omskriv til: calculations which utilities matrices and vectors, as they are often paralisable (er det et ord?)